tap_playvox/sync.py

```diff
@@ -116,13 +121,25 @@ def sync_endpoint(client,
                             record_date = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records]['date'] 
                             user_channel_records = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']]                            
-                            
-                            if(len(user_channel_records)>=1): 
+
+                            # For Worstream Agent and Schedule Metrics, only parse the data if Users data is presents
+                            if(len(user_channel_records)>=1 and stream_name == 'agent_metrics'): 
                                 user_id = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['id']
                                 record_user_firstName = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['firstName']
                                 record_user_lastName = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['lastName']
                                 records = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records][endpoint['channel_key']]
                                 parsed_user_records += 1
+
+                             # For Worstream Agent and Schedule Metrics, if no date records are present return
+                            if(len(user_channel_records)>=1 and stream_name == 'schedule_metrics'): 
+                                user_id = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['id']
+                                record_user_email = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['email']
+                                record_user_rostered_startTime = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['rosteredStartTime']
+                                record_user_rostered_endTime = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['rosteredEndTime']
+                                record_user_acutal_startTime = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['actualStartTime']
+                                record_user_actual_endTime = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]['actualEndTime']
+                                records = data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records][endpoint['channel_key']]
+                                parsed_user_records += 1
```

Comment:
you're accessing the same base object `data[endpoint['data_key']][endpoint['metric_key']][parsed_date_records][endpoint['user_key']][parsed_user_records]`. can you set this to a variable?

Reply:
Yes! Good idea, let me make that change

Reply:
Change made